From dd5e5f29a89e3f556d55767069828aa2832d6d56 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Wed, 17 Aug 2022 20:22:36 +0500
Subject: [PATCH] resize-helper: make parted not prompt for user intervention

If the device/part being resized is mounted parted asks for a confirmation
and waits for user input which does not work well in automatic resizing use
cases. We achieve this by piping a confirmation to the prompt that is displayed
for user input.

Upstream-Status: Pending

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 resize-helper | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/resize-helper b/resize-helper
index 86882f6..4fcdd45 100755
--- a/resize-helper
+++ b/resize-helper
@@ -57,6 +57,6 @@ if [ "$PART_TABLE_TYPE" = "gpt" ]; then
 	${SGDISK} -e ${DEVICE}
 	${PARTPROBE}
 fi
-${PARTED} -m ${DEVICE} u s resizepart ${PART_ENTRY_NUMBER} ${END_SIZE}
+echo "Yes" | ${PARTED} ---pretend-input-tty -m ${DEVICE} u s resizepart ${PART_ENTRY_NUMBER} ${END_SIZE}
 ${PARTPROBE}
 ${RESIZE2FS} "${ROOT_DEVICE}"
-- 
2.17.1

